### ISSUE TITLE
Strange html snippets on page

### ISSUE THREAD

**Issue by MyrionPhoenixmoon** [jekyll-octopod/jekyll-octopod#35]
I realized that since the latest Octopod update, my site includes some strange snippets of html in highlight tags:
![section](https://cloud.githubusercontent.com/assets/1800739/26281910/30d7602a-3e05-11e7-859a-8a8be2f25a47.png)

I have determined that they are the beginning of the next post_line, where the post_line doesn't get wrapped in the section tags, but instead is cut up into three pieces: The highlighted opening section tag, a div with the contents and a paragraph with the closing section tag.

**Comment by haslinger** [comment]
The sections itself are in since I took over the project (obviously not highlighted), their purpose it to be able to deep link into an episode. I never touched them, so I hope it comes from your show notes.

Code highlighting usually starts with three `, so are there any occurrences for these in your last show notes? Is the markdown file of your last episode on Github?

**Comment by haslinger** [comment]
Update: Meanwhile, I assume the origin to be connected to isso-comments.
When you open your episode via: https://swiss-sided-dice.ch/2017/03/20/gurps-dust-session-2.html#787bcb02 
everything is fine, while on the front, that should be pretty much the same, it isn't. So maybe start with comparing these two.

**Comment by MyrionPhoenixmoon** [comment]
We don't have any ` characters in our show notes, so that's not it.

That's strange, because on the front page is precisely where I don't use isso comments. I added an unless clause to post.html, checking that we're not on the front page ("/") before including it. It does suggest that it has something to do with those, however. But why does the highlighter grab the first section tag?

```html
{% if post %}
{% elsif page %}
  {% assign post = page %}
{% endif %}
<section id="{{ post.id | sha1:8 }}" class="active">
  {% include post_header.html %}
  {{ post.content }}
    {% if post.audio %}
      {% for file in post.audio %}
      <a class="btn btn-primary" href="{{ site.url }}/episodes/{{ file.last }}">Download .{{ file.first }} ({{ file.last | file_size:'./' | string_of_size }})</a>
      {% endfor %}
    {% endif %}
  <hr>
  {{ site | flattr_button:post }}
  <a href="https://twitter.com/share" class="twitter-share-button" data-url="{{ site.url }}/{{post.url }}" data-text="{{ post.title }}">Tweet</a>
  {% unless page.url == '/' %}{% include isso_thread.html %}{% endunless %}
  <hr>
</section>
```

Unfortunately for this, I keep my podcast site in a private git repo, rather than on GitHub.

**Comment by haslinger** [comment]
Maybe disabling the highlighter all together?

```
kramdown: 
  syntax_highlighter_opts:
    disable: true
```
in _config.yml should do the trick, if the original reason is Markdown, if the highlighting is triggered somehow else
```
highlighter: none
```
could be it, again in _config.yml

**Comment by MyrionPhoenixmoon** [comment]
Sorry, but both of these didn't help me.

When would highlight-rouge get triggered anyway?

**Comment by haslinger** [comment]
That's neither in jekyll-octopod nor anywhere in your application directory, but within the jekyll and the kramdown (that's the markdown parser Jekyll depends on) gems. 
Sorry, but no clue, where in there.
Typical triggers are the three ` and highlight-blocks, the docs are here: https://jekyllrb.com/docs/templates/#code-snippet-highlighting and here:
https://kramdown.gettalong.org/syntax_highlighter/rouge.html

**Comment by richardkwo** [comment]
Having the same issue here. I temporarily fixed it by commenting out the `</section>` tag at the end of `post.html`.

**Comment by haslinger** [comment]
Nice to meet you, @richardkwo! I still don't have the slightest clue how to fix it (even didn't manage to reproduce it).
If any of you could create a dummy github project/podcast where the issue occurs, I could investigate.

**Comment by JDDoesDev** [comment]
I am also having this problem.

**Comment by haslinger** [comment]
If any of you could create a dummy github project/podcast where the issue occurs, I could investigate.

**Comment by JDDoesDev** [comment]
I found the issue and it has to do with markdown and indentation.  I'm currently using `kramdown` and by fiddling around with a few of the templates, I found that the line of markdown creating the issue is in `index.md` line 13.  By removing the indent by 2 spaces everything was generated correctly.

**Comment by haslinger** [comment]
I don't see why we would have to remove the ability to have markdown here (by changing the file extension), if it really would be only a matter of irdentation.
Why not just remove the indentation? Can maybe someone else reproduce that this would fix the issue?

If there really would be an issue with kramdown, I would be more than happy to report this to @gettalong, but I highly doubt that. Since the creation of this issue, I created to more new podcasts, none of them has the issue.

Until then, my old request repeated again:
_If any of you could create a dummy github project/podcast where the issue occurs, I could investigate._

**Comment by haslinger** [comment]
Finally I ran into the issue myself! @JDDoesDev was absolutely right, that the problem is with the index.md, but we can keep it as a markdown file by changing two lines.

I would love you all (so also @MyrionPhoenixmoon @richardkwo)  to try this out: 
https://github.com/jekyll-octopod/jekyll-octopod/commit/bdb9bd0cb5631528822ff83f38be7b946f5842f6 by just adding the two `-` right before the closing angle brackets in the index.md before I release a new version, that might not fix it totally.
So no need to update, just enter the two characters in your template.

**Comment by haslinger** [comment]
OK, so now protest up to now. 
Released bugfix release 0.9.2 that includes the changes.
